padding.c

```diff
@@ -421,10 +422,11 @@ bam_hdr_t * fix_header(bam_hdr_t *old, faidx_t *fai)
 
             /* Parse the @SQ lines */
             while (cp != end) {
-                if (end-cp >= 2 && strncmp(cp, "LN", 2) == 0) {
+                if (!ln_found && end-cp >= 2 && strncmp(cp, "LN", 2) == 0) {
                     // Rewrite the length
                     char len_buf[100];
                     int tid;
+                    ln_found = 1;
                     for (tid = 0; tid < header->n_targets; tid++) {
                         // may want to hash this, but new header API incoming.
                         if (strncmp(name, header->target_name[tid], name_end - name) == 0) {
```

Comment:
When printing the replacement string, we still need to ensure that it's shorter than the original.  The best way might be to copy cp into another pointer, and move the loop at line 438 up higher so we know where the tab is before trying  to make the replacement.  Then check that the return value of sprintf() is less than or equal to the difference between the two pointers.